### PR TITLE
Convert to type:module

### PR DESCRIPTION
**Pull request by mshima**
This is a cli only application, it's not used as a lib, so it's not a breaking change.
- Convert to a lib.
- Make ejs optional.
- Make the code more linear.
- Extract the cli run to cli.js.

### CODE REVIEW

**Review by thecampagnards — COMMENTED** [review]
just a question to move buildcommand inside the cli file

**Comment by thecampagnards** [comment]
I saw your project to make github actions I created a repo there https://github.com/soprasteria/sonar-report-action using this template https://github.com/actions/hello-world-javascript-action if you want to merge in it ?
Thanks for your contributions !

File: index.js [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+import ejs from "ejs";
+import { getProxyForUrl } from "proxy-from-env";
+import got from "got";
+import hpagent from "hpagent";
+import { propertiesToJson } from "properties-file";
+import semver from 'semver';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const buildCommand = (command = new Command()) => command
```
[the remark this answers is not in the document]
Comment (mshima) [reply index.js:15]: Normally it should move to `cli.js` but I am using a custom commander to integration into GitHub Actions.
https://github.com/mshima/sonar-report-action/blob/c511dbace7eec515de8403e7e3df455efc6b4572/action/program.js#L2
This means it cannot be inside the cli script to not be called.

**Comment by mshima** [comment]
> I saw your project to make github actions I created a repo there https://github.com/soprasteria/sonar-report-action using this template https://github.com/actions/hello-world-javascript-action if you want to merge in it ?

I saw yesterday that repository, but I already had the project settled.
- I still not sure if I will generate both summaries with leak and without at the action.
- There is a missing commit at sonar-report so I am using a tag: https://github.com/mshima/sonar-report/tree/action_v0, the commit is https://github.com/mshima/sonar-report/commit/288569cf2cfc71cc540e298363030a2372158720.
- An Action must have the entire node_modules inside the repository. Using the sonar-report tag is quite simple

For those reason, I will keep the action and `sonar-report` from my own repository for now.
When I find the action stable enough, I will take a look at your action to see if it fits.
You can use my action to create your own.

**Comment by mshima** [comment]
This file generates the `action.yml` https://github.com/mshima/sonar-report-action/blob/c511dbace7eec515de8403e7e3df455efc6b4572/scripts/generate-action-yml.js.
I plan to generate the README using the same pattern.

**Pull request merged**